src/ROOTDataSource.cc

```diff
@@ -0,0 +1,401 @@
+#include "podio/ROOTDataSource.h"
+
+// STL
+#include <cstddef>
+#include <cstdio>
+#include <iostream>
+#include <filesystem>
+#include <exception>
+#include <memory>
+
+// ROOT
+#include <TFile.h>
+
+// podio
+#include <podio/Frame.h>
+#include <podio/ROOTReader.h>
+
+bool loadROOTDataSource() {
+  return true;
+}
+
+namespace podio {
+  /**
+   * \brief Construct the podio::ROOTDataSource from the provided file.
+   */
+  ROOTDataSource::ROOTDataSource(const std::string& filePath,
+                         int nEvents) : m_nSlots{1} {
+    m_filePathList.emplace_back(filePath);
+    SetupInput(nEvents);
+  }
+
+  /**
+   * \brief Construct the podio::ROOTDataSource from the provided file list.
+   */
+  ROOTDataSource::ROOTDataSource(
+      const std::vector<std::string>& filePathList,
+      int nEvents) : m_nSlots{1},
+                     m_filePathList{filePathList} {
+    SetupInput(nEvents);
+  }
+
+  /**
+   * \brief Setup input for the podio::ROOTDataSource.
+   */
+  void ROOTDataSource::SetupInput(int nEvents) {
+    // std::cout << "podio::ROOTDataSource: Constructing the source ..." << std::endl;
+
+    if (m_filePathList.empty()) {
+      throw std::runtime_error("podio::ROOTDataSource: No input files provided!");
+    }
+
+    for (const auto& filePath : m_filePathList) {
+      // Check if file exists
+      // Warning: file can be coming from web or eos
+      // if (!std::filesystem::exists(filePath)) {
+      //   throw std::runtime_error("podio::ROOTDataSource: Provided file \""
+      //                            + filePath + "\" does not exist!");
+      // }
+
+      // Check if the provided file contains required metadata
+      std::unique_ptr<TFile> inFile(TFile::Open(filePath.data(), "READ"));
+      auto metadata = inFile->Get("podio_metadata");
+      if (!metadata) {
+        throw std::runtime_error(
+            "podio::ROOTDataSource: Provided file is missing podio metadata!");
+      }
+    }
```

Comment:
This check is already done in `ROOTReader::openFile` and I think we can remove it here.

Reply:
Removed